PhysicsTools/ONNXRuntime/src/ONNXRuntime.cc

```diff
@@ -78,6 +78,35 @@ namespace cms::Ort {
 
   ONNXRuntime::~ONNXRuntime() {}
 
+  SessionOptions ONNXRuntime::defaultSessionOptions(GPUMode gpu_mode) {
+    SessionOptions sess_opts;
+    sess_opts.SetIntraOpNumThreads(1);
+    if (gpu_mode != no_gpu) {
+      // detect if there is GPU
+      int devices = 0;
+      auto status = cudaGetDeviceCount(&devices);
```

Comment:
I wonder if `CUDAService` should be used here instead for figuring our whether CUDA can be used or not?

Reply:
It should be used in the `EDModule` or directly here?

Reply:
`CUDAService` can be used directly here. (this function is called from the EDModule' s, right? e.g. from their constructor)